### PR TITLE
Add ArgMax and Fix shape errors

### PR DESCRIPTION
**Pull request by Vimalakirti**
- Add ArgMax layer.
- Fix the shape errors in ReduceMean and Squeeze for ResNet50.

### CODE REVIEW

**Comment by liliatangxy** [comment]
What were the shape errors that happened in ResNet50?

**Comment by Vimalakirti** [comment]
> What were the shape errors that happened in ResNet50?

The ReduceMean in ResNet requires computation for the mean of the last two axes. The previous implementation only supported the last dim.

The Squeeze in ResNet removes the last two dims. Our previous implementation only supported the removal of the last 1.

**Comment by Vimalakirti** [comment]
> What were the shape errors that happened in ResNet50?

Besides the ones I pushed, Reshape layer also has a similar problem (only support the case that the product of last two axes of startShape == the last axis of endShape). Will add the fix to this PR soon.

File: src/layer/reducemean.rs [hunk 1]
```diff
@@ -2,14 +2,48 @@ use crate::basic_block::*;
...
+    // Only support reducing along one or two axis
+    assert!(axes.len() == 1 || axes.len() == 2);
+    // reducing along the last axis
+    assert!(axes.iter().any(|&x| x == input_shapes[0].len() - 1));
+
+    let n = input_shapes[0].len();
+    let mut a = input_shapes[0][n - 1];
+    a = util::next_pow(a as u32) as usize;
+    let permutation = (vec![0], (0..a).collect());
+    let permute = graph.addBB(Box::new(RepeaterBasicBlock {
```
Comment (liliatangxy) [inline src/layer/reducemean.rs:44]: maybe add a comment saying what this does and that it is only used for axes = 2

**Review by liliatangxy — DISMISSED** [review]

**Review by liliatangxy — APPROVED** [review]

**Review by ddkang — APPROVED** [review]

**Pull request merged**